File: app/src/main/java/com/example/gymbuddy/view/AddExerciseActivity.java
package com.example.gymbuddy.view;

import android.animation.ObjectAnimator;
import android.app.Activity;
import android.app.AlertDialog;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.AdapterView;
import android.widget.Button;
import android.widget.EditText;
import android.widget.ImageView;
import android.widget.NumberPicker;
import android.widget.Spinner;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.example.gymbuddy.R;
import com.example.gymbuddy.helpers.Helpers;
import com.example.gymbuddy.model.Exercise;
import com.example.gymbuddy.model.Workout;
import com.google.android.material.textfield.TextInputLayout;

import java.lang.reflect.Field;
import java.util.ArrayList;

public class AddExerciseActivity extends AppCompatActivity {

    private Spinner edtSets, edtIso;
    private EditText edtExerciseName,edtMuscleGroupE,edtMuscleGroupSecondaryE;
    private EditText edtBreaks;
    private EditText edtTempo;
    private TextInputLayout tilTempo, tilMuscleGroupSecondaryE;
    private Button btnAddNewExercise;
    private ImageView imgSets, imgBreaks, imgExerciseName, imgTempo;
    private ArrayList<Exercise> allExercises;
    int position;
    private boolean isEdited;
    private boolean changeMade;
    private Workout workout;
    private Button btnSelect;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(R.style.Theme_GymBuddy);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_exercise);

//        getSupportActionBar().setBackgroundDrawable(new ColorDrawable(getResources().getColor(R.color.orange_500)));
//        setTitle("Add new exercise");

        Helpers.setupActionBar(getString(R.string.addNewExercise), "", getSupportActionBar(), this);

        edtIso = findViewById(R.id.edtIso);
        edtSets = findViewById(R.id.edtSets);
        edtExerciseName = findViewById(R.id.edtExerciseName);
        edtMuscleGroupE = findViewById(R.id.edtMuscleGroupE);
        edtMuscleGroupSecondaryE = findViewById(R.id.edtMuscleGroupSecondaryE);
        edtBreaks = findViewById(R.id.edtTxtBreaks);
        btnAddNewExercise = findViewById(R.id.btnAddNewExercise);
        imgSets = findViewById(R.id.imgSets);
        imgExerciseName = findViewById(R.id.imgExerciseName);
        imgBreaks = findViewById(R.id.imgBreaks);
        edtTempo = findViewById(R.id.edtTempo);
        imgTempo = findViewById(R.id.imgTempo);
        btnSelect = findViewById(R.id.btnSelect);
        tilTempo=findViewById(R.id.tilTempo);
        tilMuscleGroupSecondaryE=findViewById(R.id.tilMuscleGroupSecondaryE);

        edtExerciseName.requestFocus();
        InputMethodManager imm = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
        imm.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0);

        tilMuscleGroupSecondaryE.setEnabled(false);
//        edtMuscleGroupSecondaryE.getBackground().setColorFilter(Color.GRAY, PorterDuff.Mode.SRC_IN);

        NumberPicker.Formatter formatter = value -> {
            int temp = value * 5;
            return "" + temp;
        };

        btnSelect.setOnClickListener(v -> {
//                InputMethodManager imm = (InputMethodManager) getSystemService(Activity.INPUT_METHOD_SERVICE);
//                imm.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0);
            final AlertDialog.Builder d = new AlertDialog.Builder(AddExerciseActivity.this, R.style.DefaultAlertDialogTheme);
            LayoutInflater inflater = getLayoutInflater();
            View dialogView = inflater.inflate(R.layout.breaks_dialog, null);
            d.setTitle(R.string.breakDialog);
            d.setMessage(R.string.breakDialogMessage);
            d.setView(dialogView);
            d.setIcon(R.drawable.ic_timer);
            final NumberPicker numberPicker = dialogView.findViewById(R.id.dialogNumberPicker);
            numberPicker.setFormatter(formatter);
            numberPicker.setMaxValue(60);
            numberPicker.setMinValue(4);

            numberPicker.setWrapSelectorWheel(false);
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                numberPicker.setTextSize(80);
            }
            numberPicker.setValue(24);
            try {
                Field field = NumberPicker.class.getDeclaredField("mInputText");
                field.setAccessible(true);
                EditText inputText = (EditText) field.get(numberPicker);
                if (inputText != null) {
                    inputText.setVisibility(View.INVISIBLE);
                }
            } catch (Exception e) {
                e.printStackTrace();
            }

//                numberPicker.setOnValueChangedListener(new NumberPicker.OnValueChangeListener() {
//                    @Override
//                    public void onValueChange(NumberPicker numberPicker, int i, int i1) {
//                    }
//                });
            d.setPositiveButton(R.string.ok, (dialogInterface, i) -> edtBreaks.setText(String.valueOf(numberPicker.getValue() * 5)));
            d.setNegativeButton(R.string.cancel, null);
            AlertDialog alertDialog = d.create();
            alertDialog.show();
        });

        edtIso.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
            @Override
            public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
                tilTempo.setEnabled(position != 1);
            }

            @Override
            public void onNothingSelected(AdapterView<?> parent) {

            }
        });

        edtMuscleGroupE.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                tilMuscleGroupSecondaryE.setEnabled(s.length() != 0);
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        Intent intent = getIntent();
        if (intent != null) {
            if (intent.getSerializableExtra(Helpers.EXERCISES_KEY) != null) {
                allExercises = (ArrayList<Exercise>) intent.getSerializableExtra(Helpers.EXERCISES_KEY);
                position = intent.getIntExtra(Helpers.POSITION_KEY, 0);
                workout = (Workout) intent.getSerializableExtra(Helpers.WORKOUT_KEY);
                isEdited = true;
//                setTitle("Editing " + allExercises.get(position).getName());
                Helpers.setupActionBar(getString(R.string.editing) + " " + allExercises.get(position).getName(), "", getSupportActionBar(), this);
                edtExerciseName.setText(allExercises.get(position).getName());
                edtExerciseName.setSelection(edtExerciseName.getText().length());
                edtMuscleGroupE.setText(allExercises.get(position).getMuscleGroup());
                edtMuscleGroupSecondaryE.setText((allExercises.get(position).getMuscleGroupSecondary()));
                if (allExercises.get(position).getTempo() == 9999) {
                    edtTempo.setText(null);
                    tilTempo.setEnabled(false);
                    edtIso.setSelection(1);
                } else {
                    edtIso.setSelection(0);
                    if (allExercises.get(position).getTempo() == 0) {
                        edtTempo.setText(null);
                    } else {
                        edtTempo.setText(String.valueOf(allExercises.get(position).getTempo()));
                    }
                }
                edtBreaks.setText(String.valueOf(allExercises.get(position).getBreaks()));
                btnAddNewExercise.setText(R.string.save);
                switch (allExercises.get(position).getSets()) {
                    case 4:
                        edtSets.setSelection(1);
                        break;
                    case 5:
                        edtSets.setSelection(2);
                        break;
                    case 6:
                        edtSets.setSelection(3);
                        break;
                    case 7:
                        edtSets.setSelection(4);
                        break;
                    case 8:
                        edtSets.setSelection(5);
                        break;
                    default:
                        edtSets.setSelection(0);
                        break;
                }
            }
        }

        btnAddNewExercise.setOnClickListener(v -> {
            if (isEdited) {
                Exercise editedExercise=allExercises.get(position);
                if (!edtExerciseName.getText().toString().isEmpty() &&
                        !edtExerciseName.getText().toString().equals(editedExercise.getName())) {
                    changeMade = true;
                    editedExercise.setName(edtExerciseName.getText().toString());
                }
                if (!edtBreaks.getText().toString().isEmpty() && Integer.parseInt(edtBreaks.getText().toString()) != 0 &&
                        !edtBreaks.getText().toString().equals(String.valueOf(editedExercise.getBreaks()))) {
                    changeMade = true;
                    editedExercise.setBreaks(Integer.parseInt(edtBreaks.getText().toString()));
                }
                if (edtIso.getSelectedItem().toString().equals(getResources().getString(R.string.yes))) {
                    editedExercise.setTempo(9999);
                    changeMade = true;
                } else if (edtTempo.getText().toString().isEmpty()) {
                    editedExercise.setTempo(0);
                    changeMade = true;
                } else if (edtTempo.getText().toString().length() == 4) {
                    editedExercise.setTempo(Integer.parseInt(edtTempo.getText().toString()));
                    changeMade = true;

                }
                if (edtMuscleGroupE.getText().toString().isEmpty()) {
                    editedExercise.setMuscleGroup("");
                    editedExercise.setMuscleGroupSecondary("");
                    changeMade = true;
                } else {
                    if (!edtMuscleGroupE.getText().toString().equals(editedExercise.getMuscleGroup())) {
                        changeMade = true;
                        editedExercise.setMuscleGroup(edtMuscleGroupE.getText().toString());
                    }
                    if (!edtMuscleGroupSecondaryE.getText().toString().equals(editedExercise.getMuscleGroupSecondary())) {
                        changeMade = true;
                        editedExercise.setMuscleGroupSecondary(edtMuscleGroupSecondaryE.getText().toString());
                    }
                }
                if (!edtSets.getSelectedItem().toString().equals(String.valueOf(editedExercise.getSets()))) {
                    changeMade = true;
                    editedExercise.setSets(Integer.parseInt(edtSets.getSelectedItem().toString()));
                }
                Intent returnIntent = new Intent();
                setResult(Activity.RESULT_OK, returnIntent);
                if (changeMade) {
//                    Toast.makeText(this, "edited", Toast.LENGTH_SHORT).show();
                    returnIntent.putExtra(Helpers.EDITED_EXERCISE_KEY, editedExercise);
                }
                InputMethodManager imm1;
                imm1 = (InputMethodManager) getSystemService(Activity.INPUT_METHOD_SERVICE);
                imm1.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0);
                finish();
            } else {

                if (edtExerciseName.getText().toString().isEmpty()) {
                    Toast.makeText(AddExerciseActivity.this, R.string.nameIsRequired, Toast.LENGTH_SHORT).show();
                    imgExerciseName.setImageResource(R.drawable.ic_hexagon_double_vertical_red);
                    shake(imgExerciseName);
                } else if (edtBreaks.getText().toString().isEmpty()) {
                    Toast.makeText(AddExerciseActivity.this, R.string.breakTimeIsRequired, Toast.LENGTH_SHORT).show();
                    imgBreaks.setImageResource(R.drawable.ic_hexagon_double_vertical_red);
                    shake(imgBreaks);
                } else if (Integer.parseInt(edtBreaks.getText().toString()) == 0) {
                    Toast.makeText(AddExerciseActivity.this, R.string.breakTimeNotZero, Toast.LENGTH_SHORT).show();
                    imgBreaks.setImageResource(R.drawable.ic_hexagon_double_vertical_red);
                    shake(imgBreaks);
                } else if (!edtTempo.getText().toString().isEmpty() && edtTempo.getText().toString().length() < 4 && edtIso.getSelectedItem().toString().equals(getResources().getString(R.string.no))) {
                    Toast.makeText(AddExerciseActivity.this, R.string.tempoDigits, Toast.LENGTH_SHORT).show();
                    imgTempo.setImageResource(R.drawable.ic_hexagon_double_vertical_red);
                    shake(imgTempo);
                } else {
                    String exerciseName = edtExerciseName.getText().toString();
                    String muscleGroupE = edtMuscleGroupE.getText().toString();
                    String muscleGroupSecondaryE = "";
                    if (edtMuscleGroupSecondaryE.isEnabled()) {
                        muscleGroupSecondaryE = edtMuscleGroupSecondaryE.getText().toString();
                    }
                    int breaks = Integer.parseInt(edtBreaks.getText().toString());
                    int sets = Integer.parseInt(edtSets.getSelectedItem().toString());
                    int tempo = 0;
                    if (edtIso.getSelectedItem().toString().equals(getResources().getString(R.string.yes))) {
                        tempo = 9999;
                    } else if (!edtTempo.getText().toString().isEmpty()) {
                        tempo = Integer.parseInt(edtTempo.getText().toString());
                    }
                    Exercise exercise = new Exercise(5, exerciseName, sets, breaks, muscleGroupE, muscleGroupSecondaryE, tempo);
//                        Toast.makeText(AddExerciseActivity.this, exercise.toString(), Toast.LENGTH_SHORT).show();
                    Intent returnIntent = new Intent();
                    returnIntent.putExtra(Helpers.NEW_EXERCISE_KEY, exercise);
                    setResult(Activity.RESULT_OK, returnIntent);
                    InputMethodManager imm1;
                    imm1 = (InputMethodManager) getSystemService(Activity.INPUT_METHOD_SERVICE);
                    imm1.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0);
                    finish();
                }
            }
        });
    }

    private void shake(View v) {
        ObjectAnimator
                .ofFloat(v, "translationX", 0, 25, -25, 25, -25, 15, -15, 6, -6, 0)
                .setDuration(200)
                .start();
    }
}

File: app/src/main/java/com/example/gymbuddy/view/AddExerciseActivity.kt
package com.example.gymbuddy.view

import android.animation.ObjectAnimator
import android.app.AlertDialog
import android.content.DialogInterface
import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.view.inputmethod.InputMethodManager
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import com.example.gymbuddy.R
import com.example.gymbuddy.databinding.ActivityAddExerciseBinding
import com.example.gymbuddy.helpers.Helpers
import com.example.gymbuddy.model.Exercise
import com.example.gymbuddy.model.Workout
import java.util.*
import kotlin.collections.ArrayList

class AddExerciseActivity : AppCompatActivity() {
    private lateinit var allExercises: ArrayList<Exercise>
    var position = 0
    private var isEdited = false
    private var changeMade = false
    private var workout: Workout? = null
    private lateinit var binding: ActivityAddExerciseBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(R.style.Theme_GymBuddy)
        super.onCreate(savedInstanceState)
        binding = ActivityAddExerciseBinding.inflate(layoutInflater)
        setContentView(binding.root)

        Helpers.setupActionBar(getString(R.string.addNewExercise), "", supportActionBar, this)

        binding.edtExerciseName.requestFocus()
        val imm = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
        imm.toggleSoftInput(InputMethodManager.SHOW_FORCED, 0)
        binding.tilMuscleGroupSecondaryE.isEnabled = false

        val formatter = NumberPicker.Formatter { value: Int ->
            val temp = value * 5
            "" + temp
        }
        binding.btnSelect.setOnClickListener {
            val dialogView = layoutInflater.inflate(R.layout.breaks_dialog, null)
            val d = AlertDialog.Builder(this@AddExerciseActivity, R.style.DefaultAlertDialogTheme).apply {
                setTitle(R.string.breakDialog)
                setMessage(R.string.breakDialogMessage)
                setView(dialogView)
                setIcon(R.drawable.ic_timer)
            }
            val numberPicker = dialogView.findViewById<NumberPicker>(R.id.dialogNumberPicker).apply {
                setFormatter(formatter)
                maxValue = 60
                minValue = 4
                wrapSelectorWheel = false
                value = 24
            }
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
                numberPicker.textSize = 80f
            }
            try {
                val field = NumberPicker::class.java.getDeclaredField("mInputText")
                field.isAccessible = true
                val inputText = field[numberPicker] as EditText
                inputText.visibility = View.INVISIBLE
            } catch (e: Exception) {
                e.printStackTrace()
            }
            d.setPositiveButton(R.string.ok) { _: DialogInterface?, _: Int -> binding.edtTxtBreaks.setText((numberPicker.value * 5).toString()) }
            d.setNegativeButton(R.string.cancel, null)
            d.create().show()

        }

        binding.edtIso.addTextChangedListener(object: TextWatcher{
            override fun beforeTextChanged(s: CharSequence, p1: Int, p2: Int, p3: Int) {}

            override fun onTextChanged(s: CharSequence, p1: Int, p2: Int, p3: Int) {
//                if(s.toString() == resources.getString(R.string.yes)){
//                    binding.tilTempo.isEnabled=false
//                }else{
//                    binding.tilTempo.isEnabled=false
//                }
            }
            override fun afterTextChanged(s: Editable) {
                binding.tilTempo.isEnabled = s.toString() != resources.getString(R.string.yes)
            }
        })
//        binding.edtIso.onItemSelectedListener = object : OnItemSelectedListener {
//            override fun onItemSelected(parent: AdapterView<*>?, view: View, position: Int, id: Long) {
//                binding.tilTempo.isEnabled = position != 1
//            }
//
//            override fun onNothingSelected(parent: AdapterView<*>?) {}
//        }
        binding.edtMuscleGroupE.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {}
            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
                binding.tilMuscleGroupSecondaryE.isEnabled = s.isNotEmpty()
            }
            override fun afterTextChanged(s: Editable) {}
        })
        if (intent != null) {
            if (intent.getSerializableExtra(Helpers.EXERCISES_KEY) != null) {
                allExercises = intent.getSerializableExtra(Helpers.EXERCISES_KEY) as ArrayList<Exercise>? ?: ArrayList()
                position = intent.getIntExtra(Helpers.POSITION_KEY, 0)
                workout = intent.getSerializableExtra(Helpers.WORKOUT_KEY) as Workout?
                isEdited = true
                Helpers.setupActionBar(getString(R.string.editing) + " " + allExercises[position].name, "", supportActionBar, this)
                binding.edtExerciseName.setText(allExercises[position].name)
                binding.edtExerciseName.setSelection(binding.edtExerciseName.text?.length ?: 0)
                binding.edtMuscleGroupE.setText(allExercises[position].muscleGroup)
                binding.edtMuscleGroupSecondaryE.setText(allExercises[position].muscleGroupSecondary)
                if (allExercises[position].tempo == 9999) {
                    binding.edtTempo.text = null
                    binding.tilTempo.isEnabled = false
                    binding.edtIso.setText(R.string.yes)
                } else {
                    binding.edtIso.setText(R.string.no)
                    if (allExercises[position].tempo == 0) {
                        binding.edtTempo.text = null
                    } else {
                        binding.edtTempo.setText(allExercises[position].tempo.toString())
                    }
                }
                binding.edtTxtBreaks.setText(allExercises[position].breaks.toString())
                binding.btnAddNewExercise.setText(R.string.save)
                binding.edtSets.setText(allExercises[position].sets.toString())
            }
        }

        val isoArrayAdapter = ArrayAdapter(this, R.layout.dropdown_item, resources.getStringArray(R.array.isometric))
        binding.edtIso.setAdapter(isoArrayAdapter)
        val setsArrayAdapter = ArrayAdapter(this, R.layout.dropdown_item, resources.getStringArray(R.array.setsNumber))
        binding.edtSets.setAdapter(setsArrayAdapter)

        binding.btnAddNewExercise.setOnClickListener {
            if (isEdited) {
                val editedExercise = allExercises[position]
                if (binding.edtExerciseName.text.toString().isNotEmpty() &&
                    binding.edtExerciseName.text.toString() != editedExercise.name
                ) {
                    changeMade = true
                    editedExercise.name = binding.edtExerciseName.text.toString()
                }
                if (binding.edtTxtBreaks.text.toString().isNotEmpty() && binding.edtTxtBreaks.text.toString().toInt() != 0 &&
                    binding.edtTxtBreaks.text.toString() != editedExercise.breaks.toString()
                ) {
                    changeMade = true
                    editedExercise.breaks = binding.edtTxtBreaks.text.toString().toInt()
                }
                when {
                    binding.edtIso.text.toString() == resources.getString(R.string.yes) -> {
                        editedExercise.tempo = 9999
                        changeMade = true
                    }
                    binding.edtTempo.text.toString().isEmpty() -> {
                        editedExercise.tempo = 0
                        changeMade = true
                    }
                    binding.edtTempo.text.toString().length == 4 -> {
                        editedExercise.tempo = binding.edtTempo.text.toString().toInt()
                        changeMade = true
                    }
                }
                if (binding.edtMuscleGroupE.text.toString().isEmpty()) {
                    editedExercise.muscleGroup = ""
                    editedExercise.muscleGroupSecondary = ""
                    changeMade = true
                } else {
                    if (binding.edtMuscleGroupE.text.toString() != editedExercise.muscleGroup) {
                        changeMade = true
                        editedExercise.muscleGroup = binding.edtMuscleGroupE.text.toString()
                    }
                    if (binding.edtMuscleGroupSecondaryE.text.toString() != editedExercise.muscleGroupSecondary) {
                        changeMade = true
                        editedExercise.muscleGroupSecondary = binding.edtMuscleGroupSecondaryE.text.toString()
                    }
                }
                if (binding.edtSets.text.toString() != editedExercise.sets.toString()) {
                    changeMade = true
                    editedExercise.sets = binding.edtSets.text.toString().toInt()
                }
                val returnIntent = Intent()
                setResult(RESULT_OK, returnIntent)
                if (changeMade) {
                    returnIntent.putExtra(Helpers.EDITED_EXERCISE_KEY, editedExercise)
                }
                val imm1: InputMethodManager = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
                imm1.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0)
                finish()
            } else {
                if (binding.edtExerciseName.text.toString().isEmpty()) {
                    Toast.makeText(this@AddExerciseActivity, R.string.nameIsRequired, Toast.LENGTH_SHORT).show()
                    binding.imgExerciseName.setImageResource(R.drawable.ic_hexagon_double_vertical_red)
                    shake(binding.imgExerciseName)
                } else if (binding.edtTxtBreaks.text.toString().isEmpty()) {
                    Toast.makeText(this@AddExerciseActivity, R.string.breakTimeIsRequired, Toast.LENGTH_SHORT).show()
                    binding.imgBreaks.setImageResource(R.drawable.ic_hexagon_double_vertical_red)
                    shake(binding.imgBreaks)
                } else if (binding.edtTxtBreaks.text.toString().toInt() == 0) {
                    Toast.makeText(this@AddExerciseActivity, R.string.breakTimeNotZero, Toast.LENGTH_SHORT).show()
                    binding.imgBreaks.setImageResource(R.drawable.ic_hexagon_double_vertical_red)
                    shake(binding.imgBreaks)
                } else if (binding.edtTempo.text.toString().isNotEmpty() && binding.edtTempo.text.toString().length < 4 && binding.edtIso.text.toString() == resources.getString(R.string.no)) {
                    Toast.makeText(this@AddExerciseActivity, R.string.tempoDigits, Toast.LENGTH_SHORT).show()
                    binding.imgTempo.setImageResource(R.drawable.ic_hexagon_double_vertical_red)
                    shake(binding.imgTempo)
                } else {
                    val exerciseName = binding.edtExerciseName.text.toString()
                    val muscleGroupE = binding.edtMuscleGroupE.text.toString()
                    var muscleGroupSecondaryE = ""
                    if (binding.edtMuscleGroupSecondaryE.isEnabled) {
                        muscleGroupSecondaryE = binding.edtMuscleGroupSecondaryE.text.toString()
                    }
                    val breaks = binding.edtTxtBreaks.text.toString().toInt()
                    val sets = binding.edtSets.text.toString().toInt()
                    var tempo = 0
                    if (binding.edtIso.text.toString() == resources.getString(R.string.yes)) {
                        tempo = 9999
                    } else if (binding.edtTempo.text.toString().isNotEmpty()) {
                        tempo = binding.edtTempo.text.toString().toInt()
                    }
                    val exercise = Exercise(5, exerciseName, sets, breaks, muscleGroupE, muscleGroupSecondaryE, tempo)
                    val returnIntent = Intent()
                    returnIntent.putExtra(Helpers.NEW_EXERCISE_KEY, exercise)
                    setResult(RESULT_OK, returnIntent)
                    val imm1: InputMethodManager = getSystemService(INPUT_METHOD_SERVICE) as InputMethodManager
                    imm1.toggleSoftInput(InputMethodManager.HIDE_IMPLICIT_ONLY, 0)
                    finish()
                }
            }
        }
    }

    private fun shake(v: View?) {
        ObjectAnimator
            .ofFloat(v, "translationX", 0f, 25f, -25f, 25f, -25f, 15f, -15f, 6f, -6f, 0f)
            .setDuration(200)
            .start()
    }
}